internal/charmstore/migrations.go

```diff
@@ -398,19 +409,37 @@ func migrateToArchiveDownloadStatsOnly(db StoreDatabase) error {
 			return errgo.Notef(err, "cannot set the download count on entity: %v", entity.URL)
 		}
 		if entity.PromulgatedURL != nil {
-			key, err := stats.key(db, EntityStatsKey(entity.PromulgatedURL, params.StatsArchiveDownloadPromulgated), true)
-			if err != nil {
-				return errgo.Mask(err)
+			mykey := promulgatedKey {
+				series: entity.Series,
+				name: entity.Name,
 			}
-			err = incrementArchiveDownloadStats(db, key, legacyDownloadStats)
-			if err != nil {
-				return errgo.Notef(err, "cannot set the download count on entity: %v", entity.URL)
+			if e,ok := promulgatedBySeriesAndName[mykey]; !ok || entity.PromulgatedRevision > e.PromulgatedRevision {
+				promulgatedBySeriesAndName[mykey] = entity
 			}
 		}
 	}
 	if err := iter.Err(); err != nil {
 		return errgo.Notef(err, "cannot iterate through entities")
 	}
+
+	for _, promulgated := range  promulgatedBySeriesAndName {
+		legacyDownloadStats, ok := promulgated.ExtraInfo[params.LegacyDownloadStats]
+		if !ok {
+			return errgo.Newf("cannot retrieve the legacy download stats on promulgated: %v", promulgated.URL)
```

Comment:
This should be impossible as we've already got them once. Perhaps add a comment.